dpnp/dpnp_iface_statistics.py

```diff
@@ -352,69 +350,65 @@ def histogram(a, bins=10, range=None, density=None, weights=None):
+def max(a, axis=None, out=None, keepdims=False, initial=None, where=True):
     """
     Return the maximum of an array or maximum along an axis.
 
+    Returns
+    -------
+    out : dpnp.ndarray
+        Maximum of `a`.
+
     Limitations
     -----------
-    Input array is supported as :obj:`dpnp.ndarray`.
+    Input array `a` is only supported as either :class:`dpnp.ndarray` or :class:`dpctl.tensor.usm_ndarray`.
+    Parameters `out`, `where`, and `initial` are supported only with their default values.
     Otherwise the function will be executed sequentially on CPU.
-    Parameter `out` is supported only with default value ``None``.
     Input array data types are limited by supported DPNP :ref:`Data types`.
 
+    See Also
+    --------
+    :obj:`dpnp.min` : Return tha minimum of an array.
+    :obj:`dpnp.maximum` : Element-wise maximum of two arrays, propagates NaNs.
+    :obj:`dpnp.fmax` : Element-wise maximum of two arrays, ignores NaNs.
+    :obj:`dpnp.amax` : The maximum value of an array along a given axis, propagates NaNs.
+    :obj:`dpnp.nanmax` : The maximum value of an array along a given axis, ignores NaNs.
+
     Examples
     --------
     >>> import dpnp as np
     >>> a = np.arange(4).reshape((2,2))
-    >>> a.shape
-    (2, 2)
-    >>> [i for i in a]
-    [0, 1, 2, 3]
+    >>> a
+    array([[0, 1],
+           [2, 3]])
     >>> np.max(a)
-    3
+    array(3)
 
-    """
+    >>> np.max(a, axis=0)   # Maxima along the first axis
+    array([2, 3])
+    >>> np.max(a, axis=1)   # Maxima along the second axis
+    array([1, 3])
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        # Negative values in 'shape' are not allowed in input array
-        # 306-322 check on negative and duplicate axis
-        isaxis = True
-        if axis is not None:
-            if dpnp.isscalar(axis):
-                if axis < 0:
-                    isaxis = False
-            else:
-                for val in axis:
-                    if val < 0:
-                        isaxis = False
-                        break
-                if isaxis:
-                    for i in range(len(axis)):
-                        for j in range(len(axis)):
-                            if i != j:
-                                if axis[i] == axis[j]:
-                                    isaxis = False
-                                    break
-
-        if not isaxis:
-            pass
-        elif out is not None:
-            pass
-        elif keepdims:
-            pass
-        elif initial is not None:
-            pass
-        elif where is not True:
-            pass
-        else:
-            result_obj = dpnp_max(x1_desc, axis).get_pyobj()
-            result = dpnp.convert_single_elem_array_to_scalar(result_obj)
+    >>> b = np.arange(5, dtype=float)
+    >>> b[2] = np.NaN
+    >>> np.max(b)
+    array(nan)
 
-            return result
+    """
+
+    if out is not None:
+        pass
```

Comment:
Would it be a good idea to do a w/a support of `out` argument and to raise `NotImplementedError` exception when passing `initial` and `where` with not-default values? (instead of falling back to numpy)